storage/http_client.go

```diff
@@ -59,7 +59,7 @@ type httpStorageClient struct {
 
 // newHTTPStorageClient initializes a new storageClient that uses the HTTP-JSON
 // Storage API.
-func newHTTPStorageClient(ctx context.Context, opts ...storageOption) (storageClient, error) {
+func newHTTPStorageClient(ctx context.Context, opts ...storageOption) (*httpStorageClient, error) {
 	s := initSettings(opts...)
 	o := s.clientOption
 	config := newStorageConfig(o...)
```

Comment:
This seems to be repeated for both clients. Can we push it to the higher level?

I was thinking maybe:

NewClient() and NewGRPCClient()
1. Add default options (these are different for http and grpc, both check respective STORAGE_EMULATOR_HOST)
2. Get credentials if env var not set (we can change this later)
3. Resolve settings + storageConfig (we should probably merge these two? seem to be serving the same purpose. but we can leave that for later)
4. Call respective newXXXStorageClient, which now should have none of the above steps.


That way all similar steps (adding default opts, resolving opts, auth...) are at the higher level and specific things (creating gapic client, creating http client, starting metrics) are at the lower level client creation function.